modules/_labs/segmented-control/react/lib/SegmentedControl.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import {styled} from '@workday/canvas-kit-labs-react-core';
+import {borderRadius, colors} from '@workday/canvas-kit-react-core';
+import {IconButton, IconButtonProps} from '@workday/canvas-kit-labs-react-button';
+import {mouseFocusBehavior} from '@workday/canvas-kit-react-common';
+
+export interface SegmentedControlProps {
+  /**
+   * The IconButton children of the SegmentedControl (must be at least two).
+   * TODO: Add support for text children
+   */
+  children: React.ReactElement<IconButtonProps>[];
+
+  /**
+   * The value or index of the IconButton that the SegmentedControl should be toggled on to.
+   * If a string is provided, the IconButton with the corresponding value will be selected.
+   * If a number is provided, the IconButton with the corresponding index will be selected.
+   * @default 0
+   */
+  value?: string | number;
+
+  /**
+   * The function called when a button in the SegmentedControl is toggled.
+   * If the selected button has a value, that value will be passed to the callback function;
+   * otherwise, the index of the button will be passed.
+   */
+  onChange?: (value: string | number) => void;
+}
+
+const SegmentedControlContainer = styled('div')({
+  '& button': {
+    borderRadius: borderRadius.zero,
+    border: `1px solid ${colors.soap500}`,
+    borderLeft: 'none',
+    '&[aria-pressed="true"]': {
+      borderColor: colors.blueberry400,
+      '&:hover, &:focus:hover': {
+        background: colors.blueberry400,
+      },
+    },
+    '&:first-of-type': {
+      borderRadius: `${borderRadius.m} 0 0 ${borderRadius.m}`,
+      borderLeft: `1px solid ${colors.soap500}`,
+    },
+    '&:last-of-type': {
+      borderRadius: `0 ${borderRadius.m} ${borderRadius.m} 0`,
+    },
+    '&:focus': {
+      borderRadius: borderRadius.m,
+      zIndex: 1,
```

Comment:
will this by any chance affect if it's in a different z index context?

Reply:
zIndex is only relative to siblings so it should be fine 👍